lib/Quantization/Quantization.cpp

```diff
@@ -111,333 +192,146 @@ static bool canBeQuantized(const Node *node) {
   }
 
   // Make sure that all inputs are floats.
-  for (unsigned i = 0, e = node->getNumInputs(); i < e; ++i) {
-    if (node->getNthInput(i).getElementType() != ElemKind::FloatTy) {
+  for (unsigned i = 0, e = node.getNumInputs(); i < e; ++i) {
+    if (node.getNthInput(i).getElementType() != ElemKind::FloatTy) {
       return false;
     }
   }
 
   return true;
 }
 
-/// Quantize the \p node such that all floating point inputs and outputs
-/// are quantized to int8 type with some scale and offset.
-/// \returns Quantized node.
-///
-/// \param F Function which holds the non quantized \p node.
-/// \param node Node to be quantized.
-/// \param quantizedInputs Array of already quantized inputs to the result node.
-/// \param qParams Tensor quantization parameters for all outputs of the
-///        \p node.
-static Node *quantizeNode(Function *F, Node *node,
-                          llvm::MutableArrayRef<NodeValue> quantizedInputs,
-                          llvm::ArrayRef<TensorQuantizationParams> qParams) {
-  Node *quantizedNode{};
-
-  switch (node->getKind()) {
-  case Kinded::Kind::FullyConnectedNodeKind: {
-    auto *FC = cast<FullyConnectedNode>(node);
-    assert(quantizedInputs.size() == 3 && "Invalid number of inputs");
-    assert(qParams.size() == 1 && "Invalid number of quantized outputs");
-    auto QT =
-        F->getParent()->uniqueType(ElemKind::Int8QTy, FC->getResult().dims(),
-                                   qParams[0].scale, qParams[0].offset);
-    quantizedNode =
-        F->createFullyConnected(FC->getName(), quantizedInputs[0],
-                                quantizedInputs[1], quantizedInputs[2], QT);
-    break;
-  }
+Node &FunctionQuantizer::morphNode(Node &node) {
+  // FIXME: Right now, the TensorQuantizationParams only tracks one
```

Comment:
Could you create a GH issue to track this FIXME, if you haven't already?

Reply:
Sure! I was wanting for @rdzhabarov to chime in to see what is his opinion for that, but we could and should definitely track that in an issue.

Reply:
That's #1832.